druid-shell/src/gtk/mod.rs

```diff
@@ -0,0 +1,689 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! GTK-based platform support
+
+use std::any::Any;
+use std::cell::{Cell, RefCell};
+use std::ffi::c_void;
+use std::ffi::OsString;
+use std::os::raw::{c_int, c_uint};
+use std::ptr;
+use std::slice;
+use std::sync::{Arc, Mutex, Weak};
+
+use gdk::{EventKey, EventMask, ModifierType, ScrollDirection, WindowExt};
+use gtk::{
+    AccelGroup, ApplicationWindow, BoxExt, Cast, ContainerExt, GtkApplicationExt, GtkWindowExt,
+    Inhibit, WidgetExt, WidgetExtManual,
+};
+use piet_common::{Piet, RenderContext};
+use util::assert_main_thread;
+use win_main::with_application;
+
+use crate::keyboard;
+use crate::kurbo::{Point, Vec2};
+use crate::platform::dialog::{FileDialogOptions, FileDialogType};
+use crate::platform::menu::Menu;
+use crate::window::{self, Cursor, MouseButton, Text, TimerToken, WinCtx, WinHandler};
+use crate::Error;
+
+pub mod application;
+pub mod dialog;
+pub mod menu;
+pub mod util;
+pub mod win_main;
+
+#[derive(Clone, Default)]
+pub struct WindowHandle {
+    state: Weak<WindowState>,
+}
+
+/// Builder abstraction for creating new windows
+pub struct WindowBuilder {
+    handler: Option<Box<WinHandler>>,
```

Comment:
This should be `Box<dyn WinHandler>` under 2018 syntax. There are a few others of these, they should show up as compiler warnings when you build?